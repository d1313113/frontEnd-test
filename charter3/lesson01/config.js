export function generateConfig() {
  return {
    server: 'http://localhost',
    port: 80801,
    domain: 'localhost'
  };
};

export function generateConfig2() {
  return {
    server: 'http://localhost',
    port: 80801,
    domain: 'localhost',
    time: new Date()
  };
}