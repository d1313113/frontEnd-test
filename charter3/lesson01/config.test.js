import { generateConfig, generateConfig2 } from "./config";

test("测试 generateConfig 函数", () => {
  expect(generateConfig()).toEqual({
    server: "http://localhost",
    port: 80801,
    domain: "localhost"
  });

  expect(generateConfig()).toMatchSnapshot();
});

test("测试 generateConfig 函数2", () => {
  expect(generateConfig2()).toMatchSnapshot({
    time: expect.any(Date)
  });
});

test("测试 generateConfig 行内inline", () => {
  expect(generateConfig2()).toMatchInlineSnapshot(
    {
      time: expect.any(Date)
    },
    `
    Object {
      "domain": "localhost",
      "port": 80801,
      "server": "http://localhost",
      "time": Any<Date>,
    }
  `
  );
});
