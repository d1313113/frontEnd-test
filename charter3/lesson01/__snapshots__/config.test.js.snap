// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 generateConfig 函数 1`] = `
Object {
  "domain": "localhost",
  "port": 80801,
  "server": "http://localhost",
}
`;

exports[`测试 generateConfig 函数2 1`] = `
Object {
  "domain": "localhost",
  "port": 80801,
  "server": "http://localhost",
  "time": Any<Date>,
}
`;
